benchmarking/smartseq2/R/analysis_functions.R

```diff
@@ -0,0 +1,327 @@
+    formula = my.formula,
+    eq.with.lhs = "italic(hat(y))~`=`~",
+    aes(
+      label = paste(..eq.label.., ..rr.label.., sep = "~~~"),
+      size = 8
+    ),
+    parse = TRUE,
+    size = 8
+  )
+  p <-
+    p + stat_fit_glance(
+      method = 'lm',
+      method.args = list(formula = my.formula),
+      geom = 'text',
+      aes(
+        label = paste("P-value:", fpval , sep = ""),
+        size = 12
+      ),
+      label.x.npc = 'left',
+      label.y.npc = 0.85,
+      size = 8
+    )
+  p <- p + theme(legend.position = "top")
+  p <- addTheme(p)
+  p <- p + xlab(paste('Base')) + ylab(paste('Updated'))
+  p <-
+    p + geom_abline(
+      data = coefs,
+      mapping = aes(
+        slope = s,
+        intercept = ic,
+        linetype = factor(tl),
+        color = factor(tl)
+      )
+    )
+  return(list(
+    'p' = p,
+    'beta' = beta,
+    'a' = a,
+    'r2' = r2,
+    'fpval' = fpval
+  ))
+}
+# plot violin plot
+# run mean comparison test
+plotBox <- function(df) {
+  mz <- melt(df)
+  p <-
+    ggviolin(
+      mz,
+      x = 'variable',
+      y = 'value',
+      color = "variable",
+      fill = "variable",
+      add = "Violin plot",
+      palette = "jco",
+      add.params = list(fill = "white")
+    )
+  p <-
+    p + stat_compare_means(comparisons = list(c('Base', 'Updated')), label = "p.signif") +
+    stat_compare_means(size = 10) + # Add pairwise comparisons p-value
+    grids(linetype = "dashed") +
+    theme_bw()
+  p$layers[[2]]$aes_params$size = 1
+  p$layers[[2]]$aes_params$textsize <- 10
+  p <- addTheme(p)
+  stat <- wilcox.test(df[, 'Base'], df[, 'Updated'])
+  return(list('p' = p, 'pval' = stat$p.value))
+}
+# plot histogram/density plot
+plothist <- function(df) {
+  # melt dataframe
+  mz <- melt(df)
+  # group mean values
+  mu <- apply(df, 2, mean)
+  density.p <- ggdensity(
+    mz,
+    x = "value",
+    add = "mean",
+    rug = TRUE,
+    color = "variable",
+    fill = "variable"
+  )
+  density.p <- density.p + grids(linetype = "dashed") + theme_bw()
+  density.p <- addTheme(density.p)
+  return(list('p' = density.p))
+}
+# ks test
+plotKS <- function(df) {
+  mz <- melt(df)
+  x <- df$Base
+  y <- df$Updated
+  ks <- ks.test(x, y)
+  # run cdf
+  cdf1 <- ecdf(x)
+  cdf2 <- ecdf(y)
+  # find max distance D-stats
+  minMax <- seq(min(x, y), max(x, y), length.out = length(x))
+  x0 <-
+    minMax[which(abs(cdf1(minMax) - cdf2(minMax)) == max(abs(cdf1(minMax) - cdf2(minMax))))]
+  y0 <- cdf1(x0)
+  y1 <- cdf2(x0)
+  # plot ecdf
+  ks.p <-
+    ggplot(mz, aes(x = value, group = variable, color = variable)) +
+    stat_ecdf(size = 1) +
+    theme_bw() +
+    theme(legend.position = "top") +
+    ylab("ECDF")
+  # do KS test
+  dtable <-
+    data.frame(
+      'test' = 'K-S',
+      'D-stats' = round(ks$statistic, 4),
+      'P-value' = convert2star(ks$p.value)
+    )
+  dtable.p <-
+    ggtexttable(dtable,
+                row = NULL,
+                theme = ttheme('mBlue', base_size = 18))
```

Comment:
Shouldn't this be `ttheme -> theme`? How did this work?

Reply:
it stands for `table theme` so should be `ttheme()`

Reply:
Ah ok. This reminds me, could you put the loading library lines in this file? That way you don't need to load the libraries ( and know which ones to load) ahead of time if you're using these functions?